packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -123,6 +123,19 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             chunkStart = this.state.pos;
             break;
 
+          case charCodes.greaterThan:
+          case charCodes.rightCurlyBrace:
+            if (process.env.BABEL_8_BREAKING) {
+              const htmlEntity =
+                ch === charCodes.rightCurlyBrace ? "&rbrace;" : "&gt;";
+              const char = this.input[this.state.pos];
+              this.raise(
+                this.state.pos,
+                `Unexpected token \`${char}\`. Did you mean \`${htmlEntity}\` or \`{'${char}'}\`?`,
+              );
+            }
+          /* falls through */
+
           default:
```

Comment:
`make prepublish-build` generates this code:
```js
        case 62:
        case 125:

        default:
```

while `BABEL_8_BREAKING=true make prepublish-build` generates this: (with the first commit of #12447 that fixes the boolean logic in the build process)
```js

        case 62:
        case 125:
          {
            const htmlEntity = ch === 125 ? "&rbrace;" : "&gt;";
            const char = this.input[this.state.pos];
            this.raise(this.state.pos, `Unexpected token \`${char}\`. Did you mean \`${htmlEntity}\` or \`{'${char}'}\`?`);
          }

        default:
```